### ISSUE TITLE
Error on Yowsup Echoclient : “InvalidMessage or KeyId for 91XXXXXXXXXX, going to send a retry”

### ISSUE THREAD

**Issue by drashtived** [tgalal/yowsup#2211]
I have set up an Echoclient of Yowsup. However, randomly I receive an error "InvalidMessage or KeyId for 91XXXXXXXXXX, going to send a retry" while reading incoming message. After a certain period of time it shows "Storing prekeys". Once this is done the Echoclient works again.
I have already tried the following solutions referring to previous posts: 
1. Installing python-axolotl from https://github.com/tgalal/python-axolotl/tree/develop 
2. Changing PROP_IDENTITY_AUTOTRUST at axolotl/props.py to TRUE 
3. Installing develop branch of the the Yowsup repository
But nothing seems to resolve this issue. Hence,posting as a new issue. If anyone has faced and resolved this issue, please help.

**Comment by AragurDEV** [comment]
This is caused by keys changing.
So your client is frequently logging out and logging in into WhatsApp or your saved atoxtldb is damaged.
Just try to delte it manually. (Delete your ~/.yowsup directory)

**Comment by pabloruan0710** [comment]
Fixed for me!

https://stackoverflow.com/questions/45736935/error-on-yowsup-echoclient-invalidmessage-or-keyid-for-91xxxxxxxxxx-going-to

**Comment by suvadipg** [comment]
Still getting the same issue while receiving messages. Any idea how to resolve it?

**Comment by jcruzp** [comment]
> it happens to me too.
> it keeps rolling the same message "InvalidMessage or InvalidKeyIdException".
> Nothing else to do appart from deleting the keys manually and login again to reload them.
> That works.

Dear HGC2002 
How have you deleted the keys?
Thanks

**Comment by tgalal** [comment]
fixed

**Comment by Neolysion** [comment]
I'm having the same problem right now. 
Tried it on yowsup 3.2.3 and on the latest version from github.
Removing the config folder with the axolotl.db didn't help either. 
I can however send messages and the recipient will receive the correctly.

**Comment by tgalal** [comment]
@Neolysion Describe the problem you are having and not that seeing "InvalidMessage or KeyId is not a problem"

**Comment by Neolysion** [comment]
I think yowsup is not able to decrypt messages. yowsup-cli doesn't show any incoming messages and in my code the onMessage method never gets called But sending messages works just fine. 
I attached the debug log from yowsup-cli if that helps.

```
D 2019-05-27 15:45:20,931 yowsup.axolotl.manager - level_prekeys(force=False)
D 2019-05-27 15:45:20,945 yowsup.axolotl.manager - len(pending_prekeys) = 812
D 2019-05-27 15:45:20,945 yowsup.axolotl.manager - load_unsent_prekeys
D 2019-05-27 15:45:20,946 yowsup.layers.noise.layer - Received auth event
D 2019-05-27 15:45:20,946 yowsup.env.env - Env not set, setting it to android
D 2019-05-27 15:45:20,946 yowsup.env.env - Current env changed to android
D 2019-05-27 15:45:20,947 yowsup.layers.noise.layer - Performing handshake [username= 41XXXXXXX36, passive=False]
D 2019-05-27 15:45:20,947 yowsup.layers.noise.layer - Starting handshake worker
D 2019-05-27 15:45:21,223 yowsup.layers.logger.layer - rx:
<success t="1558964721" props="25" location="atn" creation="1298220271">
</success>

D 2019-05-27 15:45:21,228 yowsup.layers.protocol_iq.layer - starting ping thread.
Auth: Logged in!
D 2019-05-27 15:45:21,231 yowsup.layers.logger.layer - rx:
<ib from="s.whatsapp.net">
<edge_routing>
<routing_info>

HEX3:b'08050802'
</routing_info>
<dns_domain>
fb
HEX3:b'6662'
</dns_domain>
</edge_routing>
</ib>
[connected]:
D 2019-05-27 15:45:21,253 yowsup.layers.protocol_ib.layer - ignoring edge_routing ib node for now
D 2019-05-27 15:45:21,260 yowsup.layers.logger.layer - rx:
<ib from="s.whatsapp.net">
<dirty type="groups" timestamp="1558964721">
</dirty>
</ib>

D 2019-05-27 15:45:21,267 yowsup.layers.logger.layer - rx:
<message from="41XXXXXXX79@s.whatsapp.net" type="text" id="3AE1DCB19D2FC566DC18" offline="5" t="1558962396" notify="Biljana">
<enc count="1" v="2" type="pkmsg">
3!ydÕ½¾Þq¦ÝNE       Ep01Iz!1âEBï wÿM!ÅîÑñÝXÈû²X[k4«;¢@z¤cpjClå7"B3
!ÿBvØEÅØ¨!Áò¤¿Tùº6ýLfà:'|öè¡% "¨äðÀ¥ùñ[ÑÁÐÃä¥mz%(£Ä!0&
HEX3:b'33122105791964d5bd96bede71a6dd4e450945703031497a213196e24542ef208a77ff4d1a210514c5eed1f1dd1f58c8fbb203585b066b34ab3ba2407aa46370886a436ce58e372242330a2105ff4276d845c5d8a821c18615f2a4bf54f9ba36fd4c66129de03a277cf6e8a125100118002210a8068ae4f07fc0a58af9f15bd1c19ed0c3e4a558086d7a2528a389c4213026'
</enc>
</message>

D 2019-05-27 15:45:21,280 yowsup.axolotl.manager - decrypt_pkmsg(senderid=41XXXXXXX79, data=(omitted), unpad=True)
D 2019-05-27 15:45:21,282 yowsup.axolotl.manager - get_session_cipher(recipientid=41XXXXXXX79)
W 2019-05-27 15:45:21,283 yowsup.layers.axolotl.layer_receive - InvalidMessage or KeyId for 41XXXXXXX79, going to send a retry
D 2019-05-27 15:45:21,284 yowsup.layers.logger.layer - tx:
<receipt id="3AE1DCB19D2FC566DC18" to="41XXXXXXX79@s.whatsapp.net" type="retry">
<retry count="1" id="3AE1DCB19D2FC566DC18" v="1" t="1558962396">
</retry>
<registration>
b'z\x8cB\x9f'
HEX:b'7a8c429f'
</registration>
</receipt>

D 2019-05-27 15:45:21,295 yowsup.layers.logger.layer - rx:
<message from="41XXXXXXX79@s.whatsapp.net" type="text" id="3AE1DCB19D2FC566DC18" offline="5" t="1558962397" notify="Biljana">
<enc count="1" v="2" type="pkmsg">
3!ydÕ½¾Þq¦ÝNE       Ep01Iz!1âEBï wÿM!ÅîÑñÝXÈû²X[k4«;¢@z¤cpjClå7"B3
ö2Ï¤«ºÙ|´Jæ}(£Ä!0&ýLfà:'|öè¡% "Å¬%1Ý¡ô
HEX3:b'33122105791964d5bd96bede71a6dd4e450945703031497a213196e24542ef208a77ff4d1a210514c5eed1f1dd1f58c8fbb203585b066b34ab3ba2407aa46370886a436ce58e372242330a2105ff4276d845c5d8a821c18615f2a4bf54f9ba36fd4c66129de03a277cf6e8a125100218002210c5ac250631dda190f40df632cfa4ab9fbad97c96b44ae67d28a389c4213026'
</enc>
</message>

D 2019-05-27 15:45:21,305 yowsup.axolotl.manager - decrypt_pkmsg(senderid=41XXXXXXX79, data=(omitted), unpad=True)
D 2019-05-27 15:45:21,306 yowsup.axolotl.manager - get_session_cipher(recipientid=41XXXXXXX79)
W 2019-05-27 15:45:21,307 yowsup.layers.axolotl.layer_receive - InvalidMessage or KeyId for 41XXXXXXX79, going to send a retry
D 2019-05-27 15:45:21,308 yowsup.layers.logger.layer - tx:
<receipt id="3AE1DCB19D2FC566DC18" to="41XXXXXXX79@s.whatsapp.net" type="retry">
<retry count="1" id="3AE1DCB19D2FC566DC18" v="1" t="1558962397">
</retry>
<registration>
b'z\x8cB\x9f'
HEX:b'7a8c429f'
</registration>
</receipt>

D 2019-05-27 15:45:21,317 yowsup.layers.logger.layer - rx:
<message from="41XXXXXXX79@s.whatsapp.net" type="text" id="3AE1DCB19D2FC566DC18" offline="5" t="1558962397" notify="Biljana">
<enc count="1" v="2" type="pkmsg">
3!ydÕ½¾Þq¦ÝNE       Ep01Iz!1âEBï wÿM!ÅîÑñÝXÈû²X[k4«;¢@z¤cpjClå7"R3
!ÿBvØEÅØ¨!Áò¤¿Tùº6ýLfà:'|öè¡% " Â[cqy¹\MÓ¿ßÀÐLÌDä­©f(£Ä!0&
HEX3:b'33122105791964d5bd96bede71a6dd4e450945703031497a213196e24542ef208a77ff4d1a210514c5eed1f1dd1f58c8fbb203585b066b34ab3ba2407aa46370886a436ce58e372252330a2105ff4276d845c5d8a821c18615f2a4bf54f9ba36fd4c66129de03a277cf6e8a12510031800222003c25b6371160479b95c8c4d16d31182bf13df07c0048e1595d0994ccc44e4ad968ca91ba366940528a389c4213026'
</enc>
</message>

D 2019-05-27 15:45:21,323 yowsup.axolotl.manager - decrypt_pkmsg(senderid=41XXXXXXX79, data=(omitted), unpad=True)
D 2019-05-27 15:45:21,325 yowsup.axolotl.manager - get_session_cipher(recipientid=41XXXXXXX79)
W 2019-05-27 15:45:21,326 yowsup.layers.axolotl.layer_receive - InvalidMessage or KeyId for 41XXXXXXX79, going to send a retry
D 2019-05-27 15:45:21,327 yowsup.layers.logger.layer - tx:
<receipt id="3AE1DCB19D2FC566DC18" to="41XXXXXXX79@s.whatsapp.net" type="retry">
<retry count="1" id="3AE1DCB19D2FC566DC18" v="1" t="1558962397">
</retry>
<registration>
b'z\x8cB\x9f'
HEX:b'7a8c429f'
</registration>
</receipt>

D 2019-05-27 15:45:21,334 yowsup.layers.logger.layer - rx:
<message from="41XXXXXXX79@s.whatsapp.net" type="text" id="3AE1DCB19D2FC566DC18" offline="5" t="1558962398" notify="Biljana">
<enc count="1" v="2" type="pkmsg">
3!ydÕ½¾Þq¦ÝNE       Ep01Iz!1âEBï wÿM!ÅîÑñÝXÈû²X[k4«;¢@z¤cpjClå7"B3
e^z7aÏU;jØ1(£Ä!0&ýLfà:'|öè¡% "»ºhBs2¿W
HEX3:b'33122105791964d5bd96bede71a6dd4e450945703031497a213196e24542ef208a77ff4d1a210514c5eed1f1dd1f58c8fbb203585b066b34ab3ba2407aa46370886a436ce58e372242330a2105ff4276d845c5d8a821c18615f2a4bf54f9ba36fd4c66129de03a277cf6e8a125100418002210bbba1f68427332bf570d655e7a376196cf553b6a06d8319328a389c4213026'
</enc>
</message>

D 2019-05-27 15:45:21,342 yowsup.axolotl.manager - decrypt_pkmsg(senderid=41XXXXXXX79, data=(omitted), unpad=True)
D 2019-05-27 15:45:21,342 yowsup.axolotl.manager - get_session_cipher(recipientid=41XXXXXXX79)
W 2019-05-27 15:45:21,343 yowsup.layers.axolotl.layer_receive - InvalidMessage or KeyId for 41XXXXXXX79, going to send a retry
D 2019-05-27 15:45:21,347 yowsup.layers.logger.layer - tx:
<receipt id="3AE1DCB19D2FC566DC18" to="41XXXXXXX79@s.whatsapp.net" type="retry">
<retry count="1" id="3AE1DCB19D2FC566DC18" v="1" t="1558962398">
</retry>
<registration>
b'z\x8cB\x9f'
HEX:b'7a8c429f'
</registration>
</receipt>

D 2019-05-27 15:45:21,351 yowsup.layers.logger.layer - rx:
<message from="41XXXXXXX79@s.whatsapp.net" type="text" id="3AE1DCB19D2FC566DC18" offline="5" t="1558962398" notify="Biljana">
<enc count="1" v="2" type="pkmsg">
3!ydÕ½¾Þq¦ÝNE       Ep01Iz!1âEBï wÿM!ÅîÑñÝXÈû²X[k4«;¢@z¤cpjClå7"B3
!ÿBvØEÅØ¨!Áò¤¿Tùº6ýLfà:'|öè¡% "úOóåX"z £Ø`í8ÃÑ¥õX(£Ä!0&
HEX3:b'33122105791964d5bd96bede71a6dd4e450945703031497a213196e24542ef208a77ff4d1a210514c5eed1f1dd1f58c8fbb203585b066b34ab3ba2407aa46370886a436ce58e372242330a2105ff4276d845c5d8a821c18615f2a4bf54f9ba36fd4c66129de03a277cf6e8a1251005180022101e07fa4ff3e558227a2087a3d8601b91ed38c3d19ea5f55828a389c4213026'
</enc>
</message>

D 2019-05-27 15:45:21,363 yowsup.axolotl.manager - decrypt_pkmsg(senderid=41XXXXXXX79, data=(omitted), unpad=True)
D 2019-05-27 15:45:21,363 yowsup.axolotl.manager - get_session_cipher(recipientid=41XXXXXXX79)
W 2019-05-27 15:45:21,364 yowsup.layers.axolotl.layer_receive - InvalidMessage or KeyId for 41XXXXXXX79, going to send a retry
D 2019-05-27 15:45:21,365 yowsup.layers.logger.layer - tx:
<receipt id="3AE1DCB19D2FC566DC18" to="41XXXXXXX79@s.whatsapp.net" type="retry">
<retry count="1" id="3AE1DCB19D2FC566DC18" v="1" t="1558962398">
</retry>
<registration>
b'z\x8cB\x9f'
HEX:b'7a8c429f'
</registration>
</receipt>

D 2019-05-27 15:45:21,373 yowsup.layers.logger.layer - rx:
<message from="41XXXXXXX79@s.whatsapp.net" type="text" id="3AE1DCB19D2FC566DC18" offline="5" t="1558962399" notify="Biljana">
<enc count="1" v="2" type="pkmsg">
3!ydÕ½¾Þq¦ÝNE       Ep01Iz!1âEBï wÿM!ÅîÑñÝXÈû²X[k4«;¢@z¤cpjClå7"B3
!ÿBvØEÅØ¨!Áò¤¿Tùº6ýLfà:'|öè¡% "Ú ¤QëV_Ù!Ô§¶2±_D³~zÏ1(£Ä!0&
HEX3:b'33122105791964d5bd96bede71a6dd4e450945703031497a213196e24542ef208a77ff4d1a210514c5eed1f1dd1f58c8fbb203585b066b34ab3ba2407aa46370886a436ce58e372242330a2105ff4276d845c5d8a821c18615f2a4bf54f9ba36fd4c66129de03a277cf6e8a125100618002210da0500a451eb565fd921d499a7b632b15f449db37e7acf3128a389c4213026'
</enc>
</message>

D 2019-05-27 15:45:21,382 yowsup.axolotl.manager - decrypt_pkmsg(senderid=41XXXXXXX79, data=(omitted), unpad=True)
D 2019-05-27 15:45:21,382 yowsup.axolotl.manager - get_session_cipher(recipientid=41XXXXXXX79)
W 2019-05-27 15:45:21,383 yowsup.layers.axolotl.layer_receive - InvalidMessage or KeyId for 41XXXXXXX79, going to send a retry
D 2019-05-27 15:45:21,383 yowsup.layers.logger.layer - tx:
<receipt id="3AE1DCB19D2FC566DC18" to="41XXXXXXX79@s.whatsapp.net" type="retry">
<retry count="1" id="3AE1DCB19D2FC566DC18" v="1" t="1558962399">
</retry>
<registration>
b'z\x8cB\x9f'
HEX:b'7a8c429f'
</registration>
</receipt>

D 2019-05-27 15:45:21,393 yowsup.layers.logger.layer - rx:
<message from="41XXXXXXX79@s.whatsapp.net" type="text" id="3AE1DCB19D2FC566DC18" offline="5" t="1558962399" notify="Biljana">
<enc count="1" v="2" type="pkmsg">
3!ydÕ½¾Þq¦ÝNE       Ep01Iz!1âEBï wÿM!ÅîÑñÝXÈû²X[k4«;¢@z¤cpjClå7"B3
!ÿBvØEÅØ¨!Áò¤¿Tùº6ýLfà:'|öè¡% "îÆñ
¡R\'0Ä'¨cs7j·W?¼"(£Ä!0&
HEX3:b'33122105791964d5bd96bede71a6dd4e450945703031497a213196e24542ef208a77ff4d1a210514c5eed1f1dd1f58c8fbb203585b066b34ab3ba2407aa46370886a436ce58e372242330a2105ff4276d845c5d8a821c18615f2a4bf54f9ba36fd4c66129de03a277cf6e8a125100718002210eec6f10a94a1525c27300fc427a86373376ab71e573fbc2228a389c4213026'
</enc>
</message>

D 2019-05-27 15:45:21,401 yowsup.axolotl.manager - decrypt_pkmsg(senderid=41XXXXXXX79, data=(omitted), unpad=True)
D 2019-05-27 15:45:21,401 yowsup.axolotl.manager - get_session_cipher(recipientid=41XXXXXXX79)
W 2019-05-27 15:45:21,402 yowsup.layers.axolotl.layer_receive - InvalidMessage or KeyId for 41XXXXXXX79, going to send a retry
D 2019-05-27 15:45:21,405 yowsup.layers.logger.layer - tx:
<receipt id="3AE1DCB19D2FC566DC18" to="41XXXXXXX79@s.whatsapp.net" type="retry">
<retry count="1" id="3AE1DCB19D2FC566DC18" v="1" t="1558962399">
</retry>
<registration>
b'z\x8cB\x9f'
HEX:b'7a8c429f'
</registration>
</receipt>

D 2019-05-27 15:45:21,408 yowsup.layers.logger.layer - rx:
<message from="41XXXXXXX79@s.whatsapp.net" type="text" id="3AE1DCB19D2FC566DC18" offline="5" t="1558962400" notify="Biljana">
<enc count="1" v="2" type="pkmsg">
3!ydÕ½¾Þq¦ÝNE       Ep01Iz!1âEBï wÿM!ÅîÑñÝXÈû²X[k4«;¢@z¤cpjClå7"B3
!ÿBvØEÅØ¨!Áò¤¿Tùº6ýLfà:'|öè¡% "£Æ)Ï¤Ðæ¡6÷£Ù%ö mÒ&(£Ä!0&
HEX3:b'33122105791964d5bd96bede71a6dd4e450945703031497a213196e24542ef208a77ff4d1a210514c5eed1f1dd1f58c8fbb203585b066b34ab3ba2407aa46370886a436ce58e372242330a2105ff4276d845c5d8a821c18615f2a4bf54f9ba36fd4c66129de03a277cf6e8a125100818002210a3c6901e29cfa4d0e6a10f3691f7a3d925f692a06dd2260228a389c4213026'
</enc>
</message>

D 2019-05-27 15:45:21,415 yowsup.axolotl.manager - decrypt_pkmsg(senderid=41XXXXXXX79, data=(omitted), unpad=True)
D 2019-05-27 15:45:21,416 yowsup.axolotl.manager - get_session_cipher(recipientid=41XXXXXXX79)
W 2019-05-27 15:45:21,417 yowsup.layers.axolotl.layer_receive - InvalidMessage or KeyId for 41XXXXXXX79, going to send a retry
D 2019-05-27 15:45:21,417 yowsup.layers.logger.layer - tx:
<receipt id="3AE1DCB19D2FC566DC18" to="41XXXXXXX79@s.whatsapp.net" type="retry">
<retry count="1" id="3AE1DCB19D2FC566DC18" v="1" t="1558962400">
</retry>
<registration>
b'z\x8cB\x9f'
HEX:b'7a8c429f'
</registration>
</receipt>

D 2019-05-27 15:45:21,424 yowsup.layers.logger.layer - rx:
<message from="41XXXXXXX79@s.whatsapp.net" type="text" id="3AE1DCB19D2FC566DC18" offline="5" t="1558962400" notify="Biljana">
<enc count="1" v="2" type="pkmsg">
3!ydÕ½¾Þq¦ÝNE       Ep01Iz!1âEBï wÿM!ÅîÑñÝXÈû²X[k4«;¢@z¤cpjClå7"R3
¼7ÜS(£Ä!0&¤¿Tùº6ýLfà:'|öè¡%      " #{=(6#q®Æ×8È1Ù\(nS$ÂdÂâ
HEX3:b'33122105791964d5bd96bede71a6dd4e450945703031497a213196e24542ef208a77ff4d1a210514c5eed1f1dd1f58c8fbb203585b066b34ab3ba2407aa46370886a436ce58e372252330a2105ff4276d845c5d8a821c18615f2a4bf54f9ba36fd4c66129de03a277cf6e8a125100918002220237b3d0f28111a36237f8a71aec6d738c88a31d9895c286e5324c2648dc2e20d11bc0337dc0e155328a389c4213026'
</enc>
</message>

D 2019-05-27 15:45:21,434 yowsup.axolotl.manager - decrypt_pkmsg(senderid=41XXXXXXX79, data=(omitted), unpad=True)
D 2019-05-27 15:45:21,434 yowsup.axolotl.manager - get_session_cipher(recipientid=41XXXXXXX79)
W 2019-05-27 15:45:21,435 yowsup.layers.axolotl.layer_receive - InvalidMessage or KeyId for 41XXXXXXX79, going to send a retry
D 2019-05-27 15:45:21,435 yowsup.layers.logger.layer - tx:
<receipt id="3AE1DCB19D2FC566DC18" to="41XXXXXXX79@s.whatsapp.net" type="retry">
<retry count="1" id="3AE1DCB19D2FC566DC18" v="1" t="1558962400">
</retry>
<registration>
b'z\x8cB\x9f'
HEX:b'7a8c429f'
</registration>
</receipt>

D 2019-05-27 15:45:21,444 yowsup.layers.logger.layer - rx:
<message from="41XXXXXXX79@s.whatsapp.net" type="text" id="3AE1DCB19D2FC566DC18" offline="5" t="1558962401" notify="Biljana">
<enc count="1" v="2" type="pkmsg">
3!ydÕ½¾Þq¦ÝNE       Ep01Iz!1âEBï wÿM!ÅîÑñÝXÈû²X[k4«;¢@z¤cpjClå7"B3
!ÿBvØEÅØ¨!Áò¤¿Tùº6ýLfà:'|öè¡%
 "ÂóU»^¢79w©´v$ôÇ(£Ä!0&
HEX3:b'33122105791964d5bd96bede71a6dd4e450945703031497a213196e24542ef208a77ff4d1a210514c5eed1f1dd1f58c8fbb203585b066b34ab3ba2407aa46370886a436ce58e372242330a2105ff4276d845c5d8a821c18615f2a4bf54f9ba36fd4c66129de03a277cf6e8a125100a18002210c2f3551d95bb5e9b3fe87fa237398f77a9b4138c7624f4c728a389c4213026'
</enc>
</message>

D 2019-05-27 15:45:21,449 yowsup.axolotl.manager - decrypt_pkmsg(senderid=41XXXXXXX79, data=(omitted), unpad=True)
D 2019-05-27 15:45:21,450 yowsup.axolotl.manager - get_session_cipher(recipientid=41XXXXXXX79)
W 2019-05-27 15:45:21,452 yowsup.layers.axolotl.layer_receive - InvalidMessage or KeyId for 41XXXXXXX79, going to send a retry
D 2019-05-27 15:45:21,454 yowsup.layers.logger.layer - tx:
<receipt id="3AE1DCB19D2FC566DC18" to="41XXXXXXX79@s.whatsapp.net" type="retry">
<retry count="1" id="3AE1DCB19D2FC566DC18" v="1" t="1558962401">
</retry>
<registration>
b'z\x8cB\x9f'
HEX:b'7a8c429f'
</registration>
</receipt>

D 2019-05-27 15:45:21,574 yowsup.layers.logger.layer - rx:
<ack from="41XXXXXXX79@s.whatsapp.net" class="receipt" type="retry" id="3AE1DCB19D2FC566DC18">
</ack>

D 2019-05-27 15:45:21,582 yowsup.layers.logger.layer - rx:
<ack from="41XXXXXXX79@s.whatsapp.net" class="receipt" type="retry" id="3AE1DCB19D2FC566DC18">
</ack>

D 2019-05-27 15:45:21,594 yowsup.layers.logger.layer - rx:
<ack from="41XXXXXXX79@s.whatsapp.net" class="receipt" type="retry" id="3AE1DCB19D2FC566DC18">
</ack>

D 2019-05-27 15:45:21,605 yowsup.layers.logger.layer - rx:
<ack from="41XXXXXXX79@s.whatsapp.net" class="receipt" type="retry" id="3AE1DCB19D2FC566DC18">
</ack>

D 2019-05-27 15:45:21,614 yowsup.layers.logger.layer - rx:
<ack from="41XXXXXXX79@s.whatsapp.net" class="receipt" type="retry" id="3AE1DCB19D2FC566DC18">
</ack>

D 2019-05-27 15:45:21,618 yowsup.layers.logger.layer - rx:
<ack from="41XXXXXXX79@s.whatsapp.net" class="receipt" type="retry" id="3AE1DCB19D2FC566DC18">
</ack>

D 2019-05-27 15:45:21,626 yowsup.layers.logger.layer - rx:
<ack from="41XXXXXXX79@s.whatsapp.net" class="receipt" type="retry" id="3AE1DCB19D2FC566DC18">
</ack>

D 2019-05-27 15:45:21,630 yowsup.layers.logger.layer - rx:
<ack from="41XXXXXXX79@s.whatsapp.net" class="receipt" type="retry" id="3AE1DCB19D2FC566DC18">
</ack>

D 2019-05-27 15:45:21,708 yowsup.layers.logger.layer - rx:
<ack from="41XXXXXXX79@s.whatsapp.net" class="receipt" type="retry" id="3AE1DCB19D2FC566DC18">
</ack>

D 2019-05-27 15:45:21,716 yowsup.layers.logger.layer - rx:
<ack from="41XXXXXXX79@s.whatsapp.net" class="receipt" type="retry" id="3AE1DCB19D2FC566DC18">
```

**Comment by Neolysion** [comment]
I've just tried to debug it and I found that the first exception is being thrown in the axolotl keystore. The prekey property of every incoming message is 0 which causes this exception:
![image](https://user-images.githubusercontent.com/7968838/58431205-07cbb900-80ad-11e9-8d1f-5fdf9b11ed11.png) 
Could this cause the "InvalidMessage or KeyId" error to get thrown afterwards without actually decrypting the message?

**Comment by apepenkov** [comment]
for me it's thrown in site-packages\yowsup\axolotl\manager.py

![изображение](https://user-images.githubusercontent.com/39992738/63791008-2be4a380-c903-11e9-8977-1cff090318d0.png)
